src/compressed_chunk.c

```diff
@@ -216,37 +216,174 @@ size_t Compressed_DelRange(Chunk_t *chunk, timestamp_t startTs, timestamp_t endT
     return deleted_count;
 }
 
-// decompress chunk
-static Chunk *decompressChunk(const CompressedChunk *compressedChunk) {
-    assert(compressedChunk != NULL);
+// TODO: convert to template and unify with decompressChunk when moving to RUST
+// decompress chunk reverse
+static inline DomainChunk *decompressChunkReverse(const CompressedChunk *compressedChunk,
+                                                  uint64_t start,
+                                                  uint64_t end,
+                                                  DomainChunk *domainChunk,
+                                                  DomainChunk *domainChunkAux) {
+    uint64_t numSamples = compressedChunk->count;
+    uint64_t lastTS = compressedChunk->prevTimestamp;
+    Sample sample;
+    domainChunk->num_samples = 0;
+    domainChunk->rev = true;
+    if (unlikely(numSamples == 0 || end < start || compressedChunk->baseTimestamp > end ||
+                 lastTS < start)) {
+        return domainChunk;
```

Comment:
This contradicts the API in uncompressed chunk where you return NULL

Reply:
The caller function which called processChunk checks if the number of samples is 0 and if so returns NULL:
[https://github.com/RedisTimeSeries/RedisTimeSeries/blob/292e4b3f3350e2bf82df48ea3c56d75e7a9632b4/src/compressed_chunk.c#L443-L445](url)